### ISSUE TITLE
test.data.table() report

### ISSUE THREAD

**Issue by mrchypark** [Rdatatable/data.table#3312]
Click preview tab ^^^ above!

By continuing to file this new issue / feature request, I confirm I have :
1. searched the [live NEWS file](https://github.com/Rdatatable/data.table/blob/master/NEWS.md) to see if it has been fixed in dev already. If so, I tried the [latest dev version](https://github.com/Rdatatable/data.table/wiki/Installation#windows).
2. looked at the titles of all the issues in the [current milestones](https://github.com/Rdatatable/data.table/milestones) and am aware of those. (Adding new information to existing issues is very helpful and appreciated.)
3. [searched all issues](https://github.com/Rdatatable/data.table/issues) (i.e. not in a milestone yet) for similar issues to mine and will include links to them explaining why mine is different.
4. searched on [Stack Overflow's data.table tag](http://stackoverflow.com/questions/tagged/data.table) and there is nothing similar there.
5. read the [Support](https://github.com/Rdatatable/data.table/wiki/Support) and [Contributing](https://github.com/Rdatatable/data.table/wiki/Contributing) guides.
6. used tags under the labels menu on the right of this page, not text prefixes in the title.

#### Thanks! Please remove the text above and include the two items below.

> sessionInfo()
R version 3.5.2 (2018-12-20)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 17763)

Matrix products: default

locale:
[1] LC_COLLATE=Korean_Korea.949  LC_CTYPE=Korean_Korea.949    LC_MONETARY=Korean_Korea.949
[4] LC_NUMERIC=C                 LC_TIME=Korean_Korea.949    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] R.utils_2.7.0               R.oo_1.22.0                 R.methodsS3_1.7.1          
 [4] nanotime_0.2.3              xts_0.11-2                  zoo_1.8-4                  
 [7] bit64_0.9-7                 bit_1.1-14                  data.table_1.12.0          
[10] performanceEstimation_1.1.0 bindrcpp_0.2.2              caret_6.0-80               
[13] lattice_0.20-38             text2vec_0.5.1              tidytext_0.2.0             
[16] forcats_0.3.0               stringr_1.3.1.9000          dplyr_0.7.8                
[19] purrr_0.2.5                 readr_1.3.1                 tidyr_0.8.2                
[22] tibble_2.0.1                ggplot2_3.1.0               tidyverse_1.2.1            
[25] jsonlite_1.6                keras_2.2.0                 usethis_1.4.0              

loaded via a namespace (and not attached):
 [1] nlme_3.1-137          RcppCCTZ_0.2.5        fs_1.2.6              lubridate_1.7.4      
 [5] httr_1.4.0            SnowballC_0.6.0       tools_3.5.2           backports_1.1.3      
 [9] utf8_1.1.4            R6_2.3.0              rpart_4.1-13          lazyeval_0.2.1       
[13] colorspace_1.4-0      nnet_7.3-12           withr_2.1.2           tidyselect_0.2.5.9000
[17] compiler_3.5.2        cli_1.0.1.9000        rvest_0.3.2           formatR_1.5          
[21] xml2_1.2.0            scales_1.0.0          tfruns_1.4            digest_0.6.18        
[25] base64enc_0.1-3       pkgconfig_2.0.2       rlang_0.3.1           readxl_1.1.0.9000    
[29] rstudioapi_0.9.0      bindr_0.1.1           generics_0.0.2        tensorflow_1.9       
[33] ModelMetrics_1.2.0    tokenizers_0.2.1      magrittr_1.5          futile.logger_1.4.3  
[37] Matrix_1.2-15         Rcpp_1.0.0            munsell_0.5.0         fansi_0.4.0          
[41] reticulate_1.10       stringi_1.2.4         whisker_0.3-2         MASS_7.3-51.1        
[45] plyr_1.8.4            recipes_0.1.4         grid_3.5.2            mlapi_0.1.0          
[49] crayon_1.3.4          haven_1.1.2           splines_3.5.2         hms_0.4.2            
[53] zeallot_0.1.0         pillar_1.3.1          reshape2_1.4.3        codetools_0.2-15     
[57] stats4_3.5.2          futile.options_1.0.1  glue_1.3.0            packrat_0.5.0        
[61] lambda.r_1.2.3        RcppParallel_4.4.1    modelr_0.1.2          foreach_1.4.4        
[65] cellranger_1.1.0      gtable_0.2.0          assertthat_0.2.0      gower_0.1.2          
[69] prodlim_2018.04.18    broom_0.5.1           janeaustenr_0.1.5     class_7.3-15         
[73] survival_2.43-3       timeDate_3043.102     iterators_1.0.10      lava_1.6.4           
[77] ipred_0.9-8          
> test.data.table()
Running C:/Users/mrchypark/Documents/R/win-library/3.5/data.table/tests/tests.Rraw 
Running test id 168.1      Test 168.1 ran without errors but failed check that x equals y:
> x = DT[, as.data.frame(itime)] 
First 6 of 1 (type 'list'):     V1
1 <NA>
2 <NA>
3 <NA>
4 <NA>
5 <NA>
6 <NA>
> y = data.frame(V1 = as.ITime(x <- c("09:29:16", "10:42:40", "23:47:12",      "01:06:01", "11:35:34", "11:51:09"))) 
First 6 of 1 (type 'list'):         V1
1 09:29:16
2 10:42:40
3 23:47:12
4 01:06:01
5 11:35:34
6 11:51:09
Running test id 168.2      Test 168.2 ran without errors but failed check that x equals y:
> x = as.character(DT[, as.POSIXct(itime, tz = "UTC")]) 
First 6 of 6 (type 'character'): [1] NA NA NA NA NA NA
> y = paste(Sys.Date(), x) 
First 6 of 6 (type 'character'): [1] "2019-01-25 09:29:16" "2019-01-25 10:42:40" "2019-01-25 23:47:12" "2019-01-25 01:06:01"
[5] "2019-01-25 11:35:34" "2019-01-25 11:51:09"
'is.NA' value mismatch: 0 in current 6 in target
Running test id 168.3      Test 168.3 ran without errors but failed check that x equals y:
> x = as.character(DT[, as.POSIXct(idate, tz = "UTC")]) 
First 6 of 6 (type 'character'): [1] NA NA NA NA NA NA
> y = c("2011-11-18", "2011-11-18", "2011-11-18", "2011-11-19", "2011-11-19",      "2011-11-19") 
First 6 of 6 (type 'character'): [1] "2011-11-18" "2011-11-18" "2011-11-18" "2011-11-19" "2011-11-19" "2011-11-19"
'is.NA' value mismatch: 0 in current 6 in target
Running test id 1590.16      Test 1590.16 ran without errors but failed check that x equals y:
> x = forderv(c(x2, x1, x1, x2)) 
First 6 of 4 (type 'integer'): [1] 2 3 1 4
> y = INT(1, 4, 2, 3) 
First 6 of 4 (type 'integer'): [1] 1 4 2 3
Mean relative difference: 0.4
Running test id 1590.17      Test 1590.17 ran without errors but failed check that x equals y:
> x = base::order(c(x2, x1, x1, x2)) 
First 6 of 4 (type 'integer'): [1] 2 3 1 4
> y = INT(1, 4, 2, 3) 
First 6 of 4 (type 'integer'): [1] 1 4 2 3
Mean relative difference: 0.4
Running test id 1966.3      Test 1966.3 ran without errors but failed check that x equals y:
> x = list.files(path = pth, pattern = "\\.csv$") 
First 6 of 1 (type 'character'): [1] "o.csv"
> y = f 
First 6 of 1 (type 'character'): [1] "ö.csv"
1 string mismatch
Running test id 1966.4      Test 1966.4 ran without errors but failed check that x equals y:
> x = list.files(path = pth, pattern = "\\.csv$") 
First 6 of 1 (type 'character'): [1] "o.csv"
> y = f 
First 6 of 1 (type 'character'): [1] "ö.csv"
1 string mismatch
Running test id 1977.6         
10 longest running tests took 205s (56% of 364s)
      ID  time nTest
 1: 1648 32.53    91
 2: 1644 30.80    91
 3: 1642 30.67    91
 4: 1646 30.11    91
 5: 1650 25.03    91
 6: 1438 19.29   738
 7: 1643 10.22    14
 8: 1652  9.36    91
 9: 1639  8.84   143
10: 1835  8.80     1
Error in eval(exprs[i], envir) : 
  7 errors out of 8270 in 00:06:04 elapsed (00:02:38 cpu) on Fri Jan 25 01:32:47 2019. [endian==little, sizeof(long double)==16, sizeof(pointer)==8, TZ=Asia/Seoul, locale='LC_COLLATE=Korean_Korea.949;LC_CTYPE=Korean_Korea.949;LC_MONETARY=Korean_Korea.949;LC_NUMERIC=C;LC_TIME=Korean_Korea.949', l10n_info()='MBCS=TRUE; UTF-8=FALSE; Latin-1=FALSE; codepage=949']. Search inst/tests/tests.Rraw for test numbers: 168.1, 168.2, 168.3, 1590.16, 1590.17, 1966.3, 1966.4.

**Comment by jangorecki** [comment]
1590.17 is also affected by locale reported here: https://github.com/Rdatatable/data.table/issues/3004

**Comment by MichaelChirico** [comment]
At a glance at the output here, it looks like this was most related to `TZ=Asia/Seoul`. But current `master` passes under that time zone -- I seem to recall we've done some previous work to improve robustness of suite to different time zones.

I'll add a test of different `TZ` to our big suite to close this.
